server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2232,7 +2232,7 @@ private void loadVmDetailsInMapForExternalDhcpIp() {
                         VMInstanceVO vmInstance = _vmInstanceDao.findById(vmId);
 
                         // only load running vms. For stopped vms get loaded on starting
-                        if (vmInstance.getState() == State.Running) {
+                        if (vmInstance != null && vmInstance.getState() == State.Running) {
```

Comment:
@rhtyd possible to remove the instance id association from the Nic if VM doesn't exist? so that this nic is not picked next time.

Reply:
We shouldn't try to encourage people to do DB hacks and let CloudStack fix their mess, but at least let the management server come up.